articles/nodejs-frontend-backend/index.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/nodejs-frontend-backend/
+title: Node.js - Frontend or Backend?
+description: Node.js is a runtime environment for executing server-side code with higher efficiency and it presents a larger bandwidth to handle large code payloads.
+author: lalithnarayan-c
+date: 2020-08-20T00:00:00-15:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/nodejs-frontend-backend/hero.jpg
+    alt: front end and backend with Node.js
+---
+Let us start by understanding what Node.js is. Node.js is a runtime environment for executing server-side code with higher efficiency and it presents a larger bandwidth to handle large code payloads. The objective of this article is as follows-
+<!--more-->
+
+1. Understand and clarify any doubt of whether Node.js is frontend or backend application
+2. List out various applications in both frontend and backend
+3. Understand how Node.js complements other technologies in the frontend and backend
+
+The choice of the language that Ryan Dahl, the creator of Node.js, opted for is the most popular programming language, JavaScript.
+Node.js increased the overall control developers had over applications. Developers exercise a greater level of control over system calls, database calls, and network calls. Hence, we achieve better performing and feature-rich web applications. JavaScript is still the most preferred programming language across the world. StackOverflow survey results prove the same. This feat achieved by JavaScript enables faster progress for all the libraries and frameworks associated with JS, including node.js.
+
+![](/engineering-education/nodejs-frontend-backend/stackoverflowresults.jpg)
+[Image Source](https://insights.stackoverflow.com/survey/2020)
+
+<!-- Frontend -->
+### The Question: Frontend or Backend?
+
+How do we analyze the question? Node.js set out as a server-side engine. There are a massive number of libraries built for Node.js.
```

Comment:
May be just me - but I don't quite follow the second sentence - Node.js set out as a server-side engine? 
Did you mean to say Node.js set out to be a server side engine?